src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -124,7 +124,7 @@ private BoundExpression RewriteDeconstruction(BoundTupleExpression left, Convers
             {
                 rightParts = ((BoundTupleExpression)right).Arguments;
             }
-            else if (right.Kind == BoundKind.Conversion)
+            else if (right.Kind == BoundKind.Conversion && ((BoundConversion)right).Operand.Kind == BoundKind.ConvertedTupleLiteral)
```

Comment:
The design is that all BoundTupleLiteral should become BoundConvertedTupleLiteral when they undergo a conversion.
That said, there was a hole (which I'm fixing in https://github.com/dotnet/roslyn/pull/19679).
I'll stop by to discuss with you and @VSadov.

```xml
  <!-- 
  Tuple literals can convert to the target type.
  Once converted to a target type, they cannot be target-typed again. 
  The Converted tuple literal is one which has been already converted to a target type.
  Converted tuple literal always has a type.
  -->
  <Node Name="BoundConvertedTupleLiteral" Base="BoundTupleExpression">
```